iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/MergingViewIndex.java

```diff
@@ -15,18 +15,49 @@
 import java.util.stream.Stream;
 
 import com.powsybl.commons.PowsyblException;
+import com.powsybl.iidm.network.Battery;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.BusbarSection;
+import com.powsybl.iidm.network.Component;
+import com.powsybl.iidm.network.Connectable;
+import com.powsybl.iidm.network.DanglingLine;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.HvdcConverterStation;
+import com.powsybl.iidm.network.HvdcLine;
 import com.powsybl.iidm.network.Identifiable;
+import com.powsybl.iidm.network.LccConverterStation;
+import com.powsybl.iidm.network.Line;
+import com.powsybl.iidm.network.Load;
 import com.powsybl.iidm.network.Network;
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.RatioTapChanger;
+import com.powsybl.iidm.network.ShuntCompensator;
+import com.powsybl.iidm.network.StaticVarCompensator;
 import com.powsybl.iidm.network.Substation;
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.iidm.network.TieLine;
+import com.powsybl.iidm.network.TwoWindingsTransformer;
 import com.powsybl.iidm.network.VoltageLevel;
+import com.powsybl.iidm.network.VscConverterStation;
 
 /**
  * @author Thomas Adam <tadam at silicom.fr>
  */
 class MergingViewIndex {
 
     /** Local storage for adapters created */
-    private final Map<Identifiable<?>, AbstractAdapter<?>> adaptersCacheMap = new WeakHashMap<>();
+    private final Map<Identifiable<?>, AbstractAdapter<?>> identifiableAdaptersCached = new WeakHashMap<>();
+
+    private final Map<Component, AbstractAdapter<?>> componentAdaptersCached = new WeakHashMap<>();
+
+    private final Map<Terminal, AbstractAdapter<?>> terminalAdaptersCached = new WeakHashMap<>();
+
+    private final Map<PhaseTapChanger, AbstractAdapter<?>> ptcAdaptersCached = new WeakHashMap<>();
+
+    private final Map<RatioTapChanger, AbstractAdapter<?>> rtcAdaptersCached = new WeakHashMap<>();
```

Comment:
You can probably remove the "adapters" suffix from the name of all these variables. This information is carried by the type of the map.

Reply:
Done